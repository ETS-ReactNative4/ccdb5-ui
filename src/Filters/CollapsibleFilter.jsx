import './CollapsibleFilter.less';
import PropTypes from 'prop-types'
import React from 'react';

export default class CollapsibleFilter extends React.Component {
  constructor( props ) {
    super( props );
    this.state = {
      showChildren: props.showChildren
    };

    // This binding is necessary to make `this` work in the callback
    // https://facebook.github.io/react/docs/handling-events.html
    this._toggleChildDisplay = this._toggleChildDisplay.bind( this );
  }

  _toggleChildDisplay() {
    this.setState( {
      showChildren: !this.state.showChildren
    } );
  }

  componentWillReceiveProps( nextProps ) {
    this.setState( {
      showChildren: nextProps.showChildren
    } );
  }

  render() {
    let composeClasses = 'collapsible-filter';
    if ( this.props.className ) {
      composeClasses += ' ' + this.props.className;
    }

    const opened =
      <button className="a-btn a-btn__link hover"
              onClick={this._toggleChildDisplay}>
        Hide
        <span className="cf-icon cf-icon-minus-round"></span>
      </button>

    const closed =
      <button className="a-btn a-btn__link hover"
              onClick={this._toggleChildDisplay}>
        Show
        <span className="cf-icon cf-icon-plus-round"></span>
      </button>

    return (
      <section className={composeClasses}>
          <div className="layout-row">
            <h5 className="flex-all">{this.props.title}</h5>
            <div className="flex-fixed toggle">
              { this.state.showChildren ? opened : closed }
            </div>
          </div>
          <p>{this.props.desc}</p>
          { this.state.showChildren ? this.props.children : null }
      </section>
    );
  }
}

CollapsibleFilter.PropTypes = {
  showChildren: PropTypes.bool
}

CollapsibleFilter.defaultProps = {
  showChildren: true
}
