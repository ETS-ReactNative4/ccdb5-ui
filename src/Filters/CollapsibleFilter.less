@import (less) "../base.less";

.collapsible-filter {
  h5 {
    max-width: ~"calc(100% - 55px)";
    box-sizing: border-box;
    padding-right: @gutter-minimum;
  }

  .toggle {
    .a-btn__link {
      border: 0;

      color: @pacific;
    }

    .cf-icon {
      padding-left: @gutter-minimum;

      color: @pacific;
    }
  }
}