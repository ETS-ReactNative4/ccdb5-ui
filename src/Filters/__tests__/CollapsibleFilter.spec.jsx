import React from 'react';
import CollapsibleFilter from '../CollapsibleFilter';
import renderer from 'react-test-renderer';

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = renderer.create(
      <CollapsibleFilter title="foo" desc="bar" />
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

