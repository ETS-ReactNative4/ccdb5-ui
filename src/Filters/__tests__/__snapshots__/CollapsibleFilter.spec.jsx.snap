exports[`initial state renders without crashing 1`] = `
<section
  className="collapsible-filter">
  <div
    className="layout-row">
    <h5
      className="flex-all">
      foo
    </h5>
    <div
      className="flex-fixed toggle">
      <button
        className="a-btn a-btn__link hover"
        onClick={[Function]}>
        Hide
        <span
          className="cf-icon cf-icon-minus-round" />
      </button>
    </div>
  </div>
  <p>
    bar
  </p>
</section>
`;
